version.py

```diff
@@ -40,18 +40,27 @@
 
 
 def call_git_describe(prefix='', cwd='.'):
-    cmd = 'git describe --tags --match %s[0-9]*' % prefix
+    version_name_cmd = 'git describe --tags --match %s[0-9]*' % prefix
+    last_tag_cmd = 'git describe --tags --abbrev=0'
     try:
         version = subprocess.run(
-            cmd.split(),
+            version_name_cmd.split(),
             stdout=subprocess.PIPE,
             check=True,
         ).stdout.decode()
+
+        last_tag = subprocess.run(
+            last_tag_cmd.split(),
+            stdout=subprocess.PIPE,
+            check=True,
+        ).stdout.decode()
+
         version = version.strip()[len(prefix):]
-        if '-' in version:
-            version = '{}+dev{}.{}'.format(*version.split('-'))
+        last_tag = last_tag.strip()[len(prefix):]
+        if version != last_tag:
+            version = f"{last_tag}+dev{version[len(last_tag)+1:].replace('-','.')}"
         return version
-    except Exception:
+    except subprocess.CalledProcessError:
```

Comment:
:+1:

Reply:
I think that we should not catch `CalledProcessError`. This error means that the command failed, e.g. because the process has returned a non-zero exit code. I'd prefer the script to fail immediately and let me know that there's a problem rather than return `None` and proceed.

Also, as long as this function can return `None`, point 2 from #194 is not addressed.

Reply:
@cameel 
```call_git_describe() returns None when it fails for any reason. But this does not mean that RELEASE-VERSION file does not exist. It may exist and contain a valid version.```
There is actually a check that if there is any version in RELEASE-VERSION and function `call_git_describe()` returns None it won't change version what is in this file.
```   
    if version is None:
        version = release_version
    if version is None:
        raise ValueError("Cannot find the version number!")

    if version != release_version:
        with path.open("w") as f:
            f.write(version)
```

Reply:
But that's what I mean. If this happens, `RELEASE-VERSION` is not updated but the script does not fail. Which means that the caller will think that everything went fine and use the wrong version.

That's something that actually happened for us when we encountered the problem with dashes in version numbers. @bartoszbetka reported that the `RELEASE-VERSION` is not always updated when we run the script to generate it, even though it's supposed to be.

The function should fail if this happens.

Reply:
@cameel So the best way to do this will be deleting try-except part of code and if only any error will appear it will fail. 
I change it as I wrote. Check if this is ok

Reply:
Yeah, but please make sure you test it in all relevant scenarios:
- reading `RELEASE-VERSION` or generating it
- `git describe` fails or does not fail (e.g. because git is not installed)
- `RELEASE-VERSION` exists on disk or does not exist.

Please check all 8 combinations. I think removing this handler may have broken reading `RELEASE-VERSION` when git is not installed. If that's the case, more changes would be needed.

Reply:
@cameel 
Done. Now if we use function `get_version(generate_version=False)` we only read version from file exactly as you said.

Reply:
So please remove the exception handlers from `call_git_describe()` now.

Reply:
Done, exception handlers removed